lib/misc/blocks.coffee

```diff
@@ -0,0 +1,56 @@
+# TODO: docstrings
+
+scopes = require './scopes'
+
+module.exports =
+
+  getLine: (ed, l) ->
+    ed.getTextInBufferRange [[l, 0], [l, Infinity]]
+
+  isBlank: (l) -> l.match /^\s*(#.*)?$/
+  isEnd: (l) -> l.match /^end\b/
+  isCont: (l) -> l.match /^(else|elseif|catch|finally)\b/
+  isStart: (l) -> not (l.match(/^\s/) or @isBlank(l) or @isEnd(l) or @isCont(l))
```

Comment:
Cases like that seem so unidiomatic that it's probably ok not to support them. In the docs we'll just say that we support idiomatic code formats.